File: Civitas - Ruby/lib/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require './Sorpresa'

module Civitas
  class Casilla_sorpresa < Casilla
    
    def initialize(mazo,nombre)
      super(nombre)
      @mazo = mazo
    end
    
    #Recibe sorpresa jugador
    def recibe_jugador(iactual, todos)
      if(jugador_correcto(iactual, todos))
        @sorpresa = @mazo.siguiente
        informe(iactual, todos)
        @sorpresa.aplicar_a_jugador(iactual, todos)
      end
    end
    
  end
end

File: Civitas - Java/src/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author juanc
 */
public class CasillaSorpresa extends Casilla {
    private Sorpresa sorpresa;
    private MazoSorpresas mazo;
    
    //Tipo SORPRESA
    CasillaSorpresa(MazoSorpresas mazo, String nombre){
        super(nombre);
        this.mazo = mazo;
        this.nombre = nombre;
    }
    
    //Aplica al jugador la sorpresa y genera un informe
    @Override
    protected void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(iactual, todos)){
            //1:
            sorpresa = mazo.siguiente();
            //2:
            informe(iactual, todos);
            //3:
            sorpresa.aplicarAJugador(iactual, todos);
        }
    }
    
}
